src/compass/s1_cslc.py

```diff
@@ -1,25 +1,35 @@
-from compass import s1_rdr2geo, s1_geo2rdr, s1_resample_burst
+from compass import s1_rdr2geo, s1_geo2rdr, s1_resample, s1_geocode_slc
+from compass.utils.geo_runconfig import GeoRunConfig
 from compass.utils.runconfig import RunConfig
 from compass.utils.yaml_argparse import YamlArgparse
 
 
-def run(cfg):
-    # If boolean flag "is_reference" is true
-    # Run rdr2geo and archive reference burst
-    if cfg.is_reference:
-        s1_rdr2geo.run(cfg)
-    else:
-        s1_geo2rdr.run(cfg)
-        s1_resample_burst.run(cfg)
+def main():
+    parser = YamlArgparse(add_grid_type=True)
```

Comment:
I'm prefer to the putting `YamlArgparse` in `__main__` as this allows `s1_cslc.main` to be called from another function. Like in the [here](https://github.com/yunjunz/COMPASS/pull/1/commits/6932068270c549b9d0d0f32493508823458004bb).

Reply:
The YamlArgparse() is needed inside the main() to parse the command line inputs too, as the `entry_points` in `setup.cfg` points to the `main()` while running s1_cslc.py as executables. And the entry_points could not be `__main__`. 

I do like the call to main() from other functions as well. A simple solution to achieve both is: create a `run(cfg, grid_type)` and move most of the code inside main() there. Then we could call `s1_cslc.run()` in a similar way to other scripts in python. Do you like this version below?

```
def run(run_config_path: str, grid_type: str):
    if grid_type == 'radar':
        # CSLC workflow in radar coordinates
        # get a runconfig dict from command line args
        cfg = RunConfig.load_from_yaml(run_config_path, 's1_cslc_radar')

        if cfg.is_reference:
            # reference burst - run rdr2geo and archive it
            s1_rdr2geo.run(cfg)

        else:
            # secondary burst - run geo2rdr + resample
            s1_geo2rdr.run(cfg)
            s1_resample.run(cfg)

    elif grid_type == 'geo':
        # CSLC workflow in geo-coordinates
        # get a runconfig dict from command line argumens
        cfg = GeoRunConfig.load_from_yaml(run_config_path, 's1_cslc_geo')

        # run geocode_slc
        s1_geocode_slc.run(cfg)

def main():
    parser = YamlArgparse(add_grid_type=True)

    run(parser.run_config_path, parser.args.grid_type)
```

Reply:
LGTM